api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -113,6 +113,15 @@ default RewriteDataFiles sort(SortOrder sortOrder) {
     throw new UnsupportedOperationException("SORT Rewrite Strategy not implemented for this framework");
   }
 
+  /**
+   * Choose Z-ORDER as a strategy for this rewrite operation with a specified list of columns to use
+   * @param columns Columns to be used to generate Z-Values
+   * @return this for method chaining
+   */
+  default RewriteDataFiles zOrder(String... columns) {
```

Comment:
I remember in the original design zOrder is considered a part of the sort order, so it makes more sense to me for the input here to be `SortOrder` instead of `String... columns`. I guess this will be a shortcut for something like `SortOrder.builderFor(schema).zOrder().columns(a,b,c).bitWidth(10)` when that interface is out? But how do people define things like string column width to cutoff in this case?

Reply:
> in a follow up version we take some basic statistics from the data set and use that to develop our z ordering

do you have any current thoughts about how this would be determined by statistics? I always imagined some human decision is still needed to tune the tradeoffs among different configs.

Reply:
For most types my goal would be to bound and the condense the byte representation domain. So for example a column which has many values that translate to
```
{0, ....., 0, 1, X, Y, Z}
```
We want to shift to 

```
{1,x,y,z,0, ..., 0}
```

This involves finding min and max values, binning within that range and shifting left

At least those are my current thoughts